src/main/java/com/Crowdsourcing/loot/CrowdsourcingLoot.java

```diff
@@ -0,0 +1,208 @@
+		{
+			put(10693, "beginnerClueWarn");
+			put(10694, "easyClueWarn");
+			put(10695, "mediumClueWarn");
+			put(10723, "hardClueWarn");
+			put(10724, "eliteClueWarn");
+			put(10725, "masterClueWarn");
+		}
+	};
+	private static final int CLUE_WARNING_ENABLED = 0;
+
+	// Hunters' loot sacks
+	private static final String HUNTERS_LOOT_SACK_BASIC = "Hunters' loot sack (basic)";
+	private static final String HUNTERS_LOOT_SACK_ADEPT = "Hunters' loot sack (adept)";
+	private static final String HUNTERS_LOOT_SACK_EXPERT = "Hunters' loot sack (expert)";
+	private static final String HUNTERS_LOOT_SACK_MASTER = "Hunters' loot sack (master)";
+
+	// Clues
+	private static final Pattern CLUE_MESSAGE = Pattern.compile("You have a sneaking suspicion.*");
+
+	// Rogue outfit
+	private static final String ROGUE_MESSAGE = "Your rogue clothing allows you to steal twice as much loot!";
+
+	private HashMap<String, Object> createSkillMap(Skill s)
+	{
+		HashMap<String, Object> h = new HashMap<>();
+		h.put(s.getName(), client.getRealSkillLevel(s));
+		h.put("B" + s.getName(), client.getBoostedSkillLevel(s));
+		return h;
+	}
+
+	private int getRingOfWealth()
+	{
+		ItemContainer equipmentContainer = client.getItemContainer(InventoryID.EQUIPMENT);
+		if (equipmentContainer == null) {
+			return 0;
+		}
+
+		if (equipmentContainer.contains(ItemID.RING_OF_WEALTH) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I))
+		{
+			return 1;
+		}
+
+		if (equipmentContainer.contains(ItemID.RING_OF_WEALTH_1) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_2) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_3) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_4) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_5) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I1) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I2) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I3) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I4) ||
+			equipmentContainer.contains(ItemID.RING_OF_WEALTH_I5))
+		{
+			return 2;
+		}
+
+		return 0;
+	}
+
+	private HashMap<String, Object> getMetadataForLoot(String name)
+	{
+		HashMap<String, Object> metadata = new HashMap<>();
+
+		// universal metadata
+		metadata.put("ringOfWealth", getRingOfWealth());  // 0=not wearing, 1=wearing uncharged, 2=wearing charged
+
+		for (Map.Entry<Integer, String> entry : VARBITS_CA.entrySet())
+		{
+			int varbitId = entry.getKey();
+			String caTier = entry.getValue();
+			if (client.getVarbitValue(varbitId) == CA_CLAIMED)
+			{
+				metadata.put(caTier, true);
+			}
+		}
+
+		for (Map.Entry<Integer, String> entry : VARBITS_CLUE_WARNINGS.entrySet())
+		{
+			int varbitId = entry.getKey();
+			String clueTier = entry.getValue();
+			if (client.getVarbitValue(varbitId) == CLUE_WARNING_ENABLED)
+			{
+				metadata.put(clueTier, true);
+			}
+		}
+
+		if (name == null)
+		{
+			return metadata;
+		}
```

Comment:
Instead of returning early, can you wrap the remaining part of the function in an
```
if (name != null) {
...
}

return metadata;
```

that way there is only one return in the function